assignments/lambda-classes.js

```diff
@@ -1 +1,265 @@
 // CODE here for your Lambda Classes
+
+// Person class (base class)
+
+class Person {
+
+  constructor(props) {
+
+    this.name = props.name;
+    this.age = props.age;
+    this.gender = props.gender;
+    this.location = props.location;
+
+  }
+
+  speak() {
+
+    console.log(`Hello my name is ${this.name}, I am from ${this.location}`);
+
+  }
+
+}
+
+// Instructor class (child of Person)
+
+class Instructor extends Person {
+
+  constructor(props) {
+
+    super(props);
+
+    this.specialty = props.specialty;
+    this.favLanguage = props.favLanguage;
+    this.catchPhrase = props.catchPhrase;
+
+  }
+
+  demo(subject) {
+
+    console.log(`Today we are learning about ${subject}`);
+
+  }
+
+  grade(student, subject) {
+
+    console.log(`${student.name} receives a perfect score on ${subject}`);
+
+  }
+
+  changeGrade(student) {
+
+    let addOrSub = Math.random();
+
+    if (addOrSub > 0.5) {
+
+      student.grade += Math.floor(Math.random() * 10);
+
+      if (student.grade > 100)
+        student.grade = 100;
+
+    }
+
+    else {
+
+      student.grade -= Math.floor(Math.random() * 10);
+
+      if (student.grade < 1)
+        student.grade = 1;
+
+    }
+
+    console.log(`${student.name}'s new grade: ${student.grade}`);
+
+  }
+
+}
+
+  // Student class (child of Person)
+
+class Student extends Person {
+
+  constructor(props) {
+
+    super(props);
+
+    this.previousBackground = props.previousBackground;
+    this.className = props.className;
+    this.favSubjects = props.favSubjects;
+    this.grade = props.grade;
```

Comment:
You could use object destructuring here too:
```javascript
constructor(props) {
    const { previousBackground, className, favSubjects, grade } = props;
    super(props);
    this.previousBackground = previousBackground;
    this.className = className;
    this.favSubjects = favSubjects;
    this.grade = grade;
}
```
It's more applicable when you're going to be accessing an object's properties multiple times within that scope, but this is still a good way to get a sense of how to do it.